### PR TITLE
feat: mesh topology enhancement with layout system refactor

### PR DESCRIPTION
**Pull request by AustinChangLinksys**
## Summary
- Introduce layout blocks system and unify dashboard cards structure
- Enhance mesh topology view with diagnostics integration
- Refactor UI to apply consistent Block layout pattern across all feature pages
- Adjust Traffic Monitor timer options to Off/10s/30s/60s (default 10s)
- Add Article XIV Layout Composition Patterns to constitution

## Test plan
- [ ] Verify Traffic Monitor timer options display correctly (Off/10s/30s/60s)
- [ ] Confirm default refresh interval is 10s
- [ ] Test mesh topology view renders correctly
- [ ] Verify dashboard cards layout consistency
- [ ] Run `./run_tests.sh` — all functional tests pass

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: lib/page/unified_diagnostics/services/unified_diagnostics_service.dart [hunk 1]
```diff
@@ -605,20 +616,46 @@ class UnifiedDiagnosticsService {
...
-      final wired = node.backhaulMediaType.contains('Ethernet') ||
-          node.backhaulMediaType.contains('MoCA') ||
-          node.backhaulMediaType.contains('G.hn');
+      // Use backhaulLinkType if available, fallback to mediaType parsing
+      final linkType = node.backhaulLinkType.isNotEmpty
+          ? node.backhaulLinkType
+          : (node.backhaulMediaType.contains('Ethernet')
+              ? 'Ethernet'
+              : 'Wi-Fi');
+      final wired = linkType == 'Ethernet';
```
[the remark this answers is not in the document]
Comment (AustinChangLinksys) [reply lib/page/unified_diagnostics/services/unified_diagnostics_service.dart:626]: **False Positive** — This is not a bug.

The M60TB router hardware **only supports Ethernet and Wi-Fi** as backhaul media types. MoCA and G.hn are documented in the TR-181 specification as *possible* values, but they are **not supported by this hardware**.

The current implementation is correct for the actual product:
- `backhaulLinkType == 'Ethernet'` → wired
- Everything else → Wi-Fi (the only other option on this device)

The fallback logic at line 622-624 (`mediaType.contains('Ethernet')`) handles edge cases where `backhaulLinkType` is empty but `backhaulMediaType` indicates Ethernet.

Adding MoCA/G.hn detection would be dead code with no test coverage since we cannot produce those values from the device.

**Review by HankYuLinksys — APPROVED** [review]
The direction is solid and architecturally disciplined — approving with a few non-blocking notes.

**Strengths**
- The Layout Blocks system was iterated and consolidated rather than just bolted on (8 files/~850 lines → 6 files/~400 lines, unused components removed).
- Strong constitution compliance: USP errors mapped through `ServiceError` (Article XIII), Flutter/Material dependencies removed from the core layer, and SSoT consolidation for speed formatting and severity enums.
- Both qodo UI-Kit-First findings (`DeviceRow` → `AppListTile`, `NetworkBadgeWidget` → `AppBadge`) are already addressed in follow-up commits `37c929f` / `750363b`.

**Non-blocking notes**
1. **Scope is large and mixes concerns** — mesh topology enhancement, an app-wide layout refactor, and a codegen bump are coupled in one PR, which makes review and rollback harder. Fine to land here, but please consider splitting layout-system and codegen work into separate PRs going forward.
2. **No golden coverage for the visual changes** — the Block pattern is applied across nearly every feature page, yet the visual items in the test plan are unchecked and the baseline snapshots job is skipped. A golden run over the touched cards before merge would de-risk layout regressions.
3. **Constitution change is a spec change** — Article XIV (Layout Composition Patterns) needs explicit sign-off on the Block-vs-Card pattern design, not just the code.

Targeting `dev-2.5.0` (a feature integration branch) is appropriate for a PR this size. 👍
